### ISSUE TITLE
How come the compiled version has more features than the source code?

### ISSUE THREAD

**Issue by benelot** [HackerPoet/Composer#22]
I could at least spot that the compiled version has the instrument buttons. Also, would you mind releasing the model?

**Comment by HackerPoet** [comment]
Yeah, the compiled version is a C++ port of the python one with just a few small differences like the instrument button.

I usually don't release pre-trained models for a few reasons:
- My C++ framework is custom-built and the models are not really compatible with existing tools
- I don't really have time to document the data format and usage
- The original models are really large (in this case 350MBs) and I don't have a good place to host it
- There could be privacy or ownership issues depending on the data used
- I don't want the pre-trained models to be used commercially (code is MIT licenced though)

Hopefully it should not be too hard to train a similar model yourself with the code provided.

**Comment by benelot** [comment]
Thanks for the explanation, sounds all reasonable!
